imqsauth/http.go

```diff
@@ -702,20 +782,28 @@ func httpHandlerSetPassword(central *ImqsCentral, w http.ResponseWriter, r *http
 }
 
 func httpHandlerResetPasswordStart(central *ImqsCentral, w http.ResponseWriter, r *httpRequest) {
-	identity := strings.TrimSpace(r.http.URL.Query().Get("identity"))
-	code, msg := central.ResetPasswordStart(identity, false)
+	userId, getUserIdErr := getUserId(r)
+	if getUserIdErr != nil {
+		authaus.HttpSendTxt(w, http.StatusBadRequest, "Invalid userid")
+		return
+	}
+	code, msg := central.ResetPasswordStart(userId, false)
 	authaus.HttpSendTxt(w, code, msg)
 }
 
 func httpHandlerResetPasswordFinish(central *ImqsCentral, w http.ResponseWriter, r *httpRequest) {
-	identity := strings.TrimSpace(r.http.URL.Query().Get("identity"))
+	userId, getUserIdErr := getUserId(r)
+	if getUserIdErr != nil {
+		authaus.HttpSendTxt(w, http.StatusBadRequest, "Invalid userid")
+		return
+	}
 	token := r.http.Header.Get("X-ResetToken")
 	password := strings.TrimSpace(r.http.Header.Get("X-NewPassword"))
-	if identity == "" || token == "" || password == "" {
-		authaus.HttpSendTxt(w, http.StatusBadRequest, "Need identity in URL, and the two headers X-ResetToken and X-NewPassword")
+	if token == "" || password == "" {
+		authaus.HttpSendTxt(w, http.StatusBadRequest, "Need the two headers X-ResetToken and X-NewPassword in the URL")
```

Comment:
Headers != URL. In case you're not familiar with the HTTP protocol, now is a good time to discover how HTTP headers work.